client/core/core.go

```diff
@@ -606,9 +606,20 @@ func (dc *dexConnection) parseMatches(msgMatches []*msgjson.Match, checkSigs boo
 		if tracker.Trade().Sell {
 			swapRate = msgMatch.FeeRateBase
 		}
-		if !isCancel && swapRate > tracker.metaData.MaxFeeRate {
-			errs = append(errs, fmt.Sprintf("rejecting match %s for order %s because assigned rate (%d) is > MaxFeeRate (%d)",
-				msgMatch.MatchID, msgMatch.OrderID, swapRate, tracker.metaData.MaxFeeRate))
+
+		if !isCancel &&
+			swapRate > tracker.metaData.MaxFeeRate &&
+			(tracker.metaData.CustomSwapFeeRate == nil || swapRate > *tracker.metaData.CustomSwapFeeRate) {
+
+			if tracker.metaData.CustomSwapFeeRate != nil {
+				errs = append(errs,
+					fmt.Sprintf("rejecting match %s for order %s because assigned rate (%d) is > MaxFeeRate (%d) and customSwapFeeRate (%d)",
+						msgMatch.MatchID, msgMatch.OrderID, swapRate, tracker.metaData.MaxFeeRate, *tracker.metaData.CustomSwapFeeRate))
```

Comment:
Hmm, the server is free to prescribe any rate up to `MaxFeeRate`, so I don't think rejecting the match on account of `CustomSwapFeeRate` is correct.